src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -45,10 +47,11 @@ class InstanceOpFactoryImpl(
   }
 
   private[this] def inferNormalTaskOp(request: InstanceOpFactory.Request): Option[InstanceOp] = {
-    val InstanceOpFactory.Request(runSpec, offer, tasks, _) = request
+    val InstanceOpFactory.Request(runSpec, offer, instances, _) = request
+    val tasks: Seq[Task] = instances.values.flatMap(_.tasks)(collection.breakOut)
 
     new TaskBuilder(runSpec, Task.Id.forRunSpec, config, Some(appTaskProc)).
-      buildIfMatches(offer, tasks.values.flatMap(_.tasks)).map {
+      buildIfMatches(offer, tasks).map {
```

Comment:
um .. I stated it backwards. we'll be building task-groups from pods. :p

On Tue, Sep 13, 2016 at 11:50 AM, Jason Gilanfarr <notifications@github.com>
wrote:

> In src/main/scala/mesosphere/marathon/core/launcher/impl/
> InstanceOpFactoryImpl.scala
> <https://github.com/mesosphere/marathon/pull/4347#discussion_r78587868>:
>
> >
> >      new TaskBuilder(runSpec, Task.Id.forRunSpec, config, Some(appTaskProc)).
> > -      buildIfMatches(offer, tasks.values.flatMap(_.tasks)).map {
> > +      buildIfMatches(offer, tasks).map {
>
> Maybe the builder could actually be in PodSpec as lazy val toTaskGroup? On
> Tue, Sep 13, 2016 at 8:48 AM James DeFelice <notifications@github.com>
> wrote:
> … <#m_3243233936633942049_>
> In src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala
> <#4347 (comment)
> <https://github.com/mesosphere/marathon/pull/4347#discussion_r78587291>>:
> > > new TaskBuilder(runSpec, Task.Id.forRunSpec, config,
> Some(appTaskProc)). > - buildIfMatches(offer, tasks.values.flatMap(_.tasks)).map
> { > + buildIfMatches(offer, tasks).map { there's a whole new builder coming
> for building pods from task-groups … <#m_6047602239082565210_> On Tue, Sep
> 13, 2016 at 11:44 AM, Matthias Eichstedt < ***@***.***> wrote: In
> src/main/scala/mesosphere/marathon/core/launcher/impl/
> InstanceOpFactoryImpl.scala <#4347
> <https://github.com/mesosphere/marathon/pull/4347> (comment) <#4347
> (comment)
> <https://github.com/mesosphere/marathon/pull/4347#discussion_r78586578>>>:
> > > new TaskBuilder(runSpec, Task.Id.forRunSpec, config,
> Some(appTaskProc)). > - buildIfMatches(offer, tasks.values.flatMap(_.tasks)).map
> { > + buildIfMatches(offer, tasks).map { In order to match constraints for
> pods we should rather be passing instances instead of tasks, but before
> that's adjusted this is OK — You are receiving this because you authored
> the thread. Reply to this email directly, view it on GitHub <
> https://github.com/mesosphere/marathon/pull/4347/files/
> bf792261a825b89046d4d7e4de2aa3b2817b788d#r78586578>, or mute the thread <
> https://github.com/notifications/unsubscribe-auth/ACPVLKdqs-
> qQMsbMfxnzF2qhMSbluaZRks5qpsTTgaJpZM4J5a4D> . — You are receiving this
> because you were assigned. Reply to this email directly, view it on GitHub <
> https://github.com/mesosphere/marathon/pull/4347/files/
> bf792261a825b89046d4d7e4de2aa3b2817b788d#r78587291>, or mute the thread <
> https://github.com/notifications/unsubscribe-auth/
> AAweZtM48dbym5debydAhKHzGg386UYwks5qpsWcgaJpZM4J5a4D> .
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4347/files/bf792261a825b89046d4d7e4de2aa3b2817b788d#r78587868>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACPVLHJpX3etEUA3uh5GfSuAED4qUFKDks5qpsZSgaJpZM4J5a4D>
> .
>